### PR TITLE
feat: add share link button

### PR DESCRIPTION
**Pull request by nathaliabj**
### Pre requirements:
- [ ] I have followed (at least) the [PR section of the contributing guide](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#sending-a-pull-request).

### Reference:
Closes: #2076 

### Description:
This PR creates the share link button, i have not added it yet to the header as the size of this button is different to the current ones, once the [new header](#2100 ) is merged  this button will be added to it.

### Preview:
https://www.loom.com/share/bd7b663d339942a0892de85d3f4a6c45

### Scope of the changes (select all that apply):
- [x] Frontend  
- [ ] Backend  
- [ ] Common  
- [ ] E2E tests  
- [ ] Docs  
- [ ] CI/CD

### CODE REVIEW

**Review by rephus — APPROVED** [review]
I remember copying stuff to the clipboard was not this straightforward, and we even required Flash to make it happen 👴 . Now it seems compatible with most browsers, so it should be fine.. 
image

**Comment by hamzahc1** [comment]
@nathaliabj - when you click the copy link button in the preview - it looked like it made the Headway notifications do something... was that a coincidence when recording or is something happening?

**Comment by rephus** [comment]
> @nathaliabj - when you click the copy link button in the preview - it looked like it made the Headway notifications do something... was that a coincidence when recording or is something happening?

This button hasn't been added to the header, what are you testing exactly ? :thinking:

**Comment by hamzahc1** [comment]
> > @nathaliabj - when you click the copy link button in the preview - it looked like it made the Headway notifications do something... was that a coincidence when recording or is something happening?
> 
> This button hasn't been added to the header, what are you testing exactly ? 🤔

I was just watching the Loom video above^^^ :D

File: packages/frontend/src/components/ShareLinkButton/index.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React, { FC } from 'react';
+import { useApp } from '../../providers/AppProvider';
+import { ShareLink } from './ShareLinkButton.styles';
+
+const ShareLinktButton: FC = () => {
+    const { showToastSuccess } = useApp();
+
+    const onClick = () => {
+        const linkToShare = window.location.href;
```
Comment (ZeRego) [inline packages/frontend/src/components/ShareLinkButton/index.tsx:9]: We better use `import CopyToClipboard from 'react-copy-to-clipboard';` 
You can check how we use it in `packages/frontend/src/components/UserSettingsModal/InvitesPanel/index.tsx`

**Review by ZeRego — CHANGES_REQUESTED** [review]
Let's also wait for the new header PR to be merged so we can test it.

**Comment by nathaliabj** [comment]
@ZeRego  I have updated the component to use CopyToClipboard from react, however this component should only live in view mode, so waiting for this to be ready before adding it to the header

**Review by ZeRego — APPROVED** [review]

**Pull request merged**